lib/components/base-components/NormalComponent/NormalComponent_doInitialSilkscreenOverlapAdjustment.ts

```diff
@@ -0,0 +1,396 @@
+import type { NormalComponent } from "./NormalComponent"
+
+/**
+ * Adjust silkscreen reference designators for passive components to avoid overlaps
+ * This phase runs after PCB layout is complete but before final rendering
+ */
+export function NormalComponent_doInitialSilkscreenOverlapAdjustment(
+  component: NormalComponent<any>,
+): void {
+  if (component.root?.pcbDisabled) return
+  const { db } = component.root!
+
+  // Only process if this is a subcircuit or board
+  if (!component.isSubcircuit && component.componentName !== "Board") return
+
+  // Get all silkscreen texts in this component and its children that are reference designators for passives
+  const allSilkscreenTexts = db.pcb_silkscreen_text.list()
+  const passiveReferenceTexts = allSilkscreenTexts.filter((text) => {
+    // Check if this is a passive component reference designator
+    return /^(R\d+|C\d+|L\d+)$/i.test(text.text)
```

Comment:
avoid regex! Remove!!

you can have logic like this:

```
if (!component._adjustSilkscreenTextAutomatically) return
```

then make sure to set _adjustSilkscreenTextAutomatically = true inside Resistor.ts, Capacitor.ts and Inductor.ts